app/helpers/custom_fields_helper.rb

```diff
@@ -198,21 +198,25 @@ def custom_field_formats_for_select(custom_field)
       format.name == "hierarchy" && !OpenProject::FeatureDecisions.custom_field_of_type_hierarchy_active?
     end
 
-    OpenProject::CustomFieldFormat
-      .all_for_field(custom_field)
-      .sort_by(&:order)
-      .reject { |format| format.label.nil? }
-      .reject(&hierarchy_if_deactivated)
-      .map do |custom_field_format|
-        [label_for_custom_field_format(custom_field_format.name), custom_field_format.name]
-      end
+    OpenProject::CustomFieldFormat.all_for_field(custom_field)
+                                  .sort_by(&:order)
+                                  .reject { |format| format.label.nil? }
+                                  .reject(&hierarchy_if_deactivated)
+                                  .map do |custom_field_format|
+      [label_for_custom_field_format(custom_field_format.name), custom_field_format.name]
+    end
   end
 
   def label_for_custom_field_format(format_string)
-    format = OpenProject::CustomFieldFormat.find_by_name(format_string)
+    show_enterprise_text = format_string == "hierarchy" && !EnterpriseToken.allows_to?(:custom_field_hierarchies)
 
-    if format
-      format.label.is_a?(Proc) ? format.label.call : I18n.t(format.label)
-    end
+    format = OpenProject::CustomFieldFormat.find_by(name: format_string)
+    label = if format
```

Comment:
🟡 Could be rewritten to `format&.label.is_a?(Proc)`.

Reply:
rewritten with a bit of reordering...